src/modules/userCreationRequestMethods.ts

```diff
@@ -201,55 +211,20 @@ export const userCreationRequestsMethods = {
             throw new TRPCError({ code: 'BAD_REQUEST', message: 'User already exists' });
         }
 
-        const services = userCreationRequest.services as { serviceId: string; serviceName: string }[];
-
-        const email = userCreationRequest.corporateEmail || userCreationRequest.email;
-
-        if (userCreationRequest.createExternalAccount) {
-            const [surname, firstName, middleName] = userCreationRequest.name.split(' ');
-
-            const phone = services.find((service) => service.serviceName === 'Phone')?.serviceId;
-
-            if (!phone) {
-                throw new TRPCError({ code: 'BAD_REQUEST', message: 'Phone service is required' });
-            }
-
-            await externalUserMethods.create({
-                surname,
-                firstName,
-                middleName,
-                email,
-                phone,
-                login: userCreationRequest.login,
-                organizationUnitId: userCreationRequest.organizationUnitId,
-            });
-        }
-
         const acceptedRequest = await prisma.userCreationRequest.update({
             where: { id },
             data: { status: 'Approved', comment },
             include: { supervisor: true, organization: true, group: true },
         });
 
-        if (acceptedRequest.corporateEmail) {
-            const emailService = await prisma.externalService.findUnique({ where: { name: 'Email' } });
-            if (!emailService) throw new TRPCError({ code: 'BAD_REQUEST', message: 'Email service not found' });
-            services.push({ serviceName: emailService.name, serviceId: acceptedRequest.email });
-        }
+        let newUser: (User & { services: UserService[] }) | undefined;
 
-        const newUser = await prisma.user.create({
-            data: {
-                name: acceptedRequest.name,
-                email: acceptedRequest.email,
-                supervisorId: acceptedRequest.supervisor.id,
-                login: acceptedRequest.login,
-                title: userCreationRequest.title,
-                memberships: { create: { groupId: acceptedRequest.groupId } },
-                organizationUnitId: acceptedRequest.organizationUnitId,
-                services: { createMany: { data: services } },
-                workStartDate: acceptedRequest.date,
-            },
-        });
+        if (acceptedRequest.date && new Date() > acceptedRequest.date) {
+            console.log('accd', acceptedRequest.date);
+            console.log('newdara', new Date());
+            console.log(new Date() > acceptedRequest.date);
+            newUser = await userMethods.createUserFromRequest(acceptedRequest.id);
```

Comment:
I think user creation should always be through the job

Reply:
Its if request accepted after job already done and deleted. Do you suggest to delay job, if request not accepted in time?